emcc.py

```diff
@@ -1127,7 +1127,13 @@ def version_string():
   return f'emcc (Emscripten gcc/clang-like replacement + linker emulating GNU ld) {utils.EMSCRIPTEN_VERSION}{revision_suffix}'
 
 
-def parse_args(newargs):
+def parse_args(newargs):  # noqa: C901, PLR0912, PLR0915
+  """
+  Future modifications should consider refactoring to reduce complexity.
+  * The McCabe cyclomatiic complexity is currently 117 vs 10 recommended.
+  * There are currently 115 branches vs 12 recommended.
+  * There are currently 302 statements vs 50 recommended.
```

Comment:
In any case, can we not add these comments?  I'm not sure they are useful.

Reply:
I would very much like to leave these comments in place because the complexity numbers on these 5 functions are radically outside the recommended ranges.  The "Why is this bad?" section of these rules states the obvious "Functions with a high complexity are hard to understand and maintain".  The vast majority of developers do not know the `noqa` id by heart.  The purpose of these docstrings is to help both contributors and code reviewers notice that there are extraordinary maintainability issues so that careful review and robust tests need to be added around any change.  It might serve as a learning experience for both mentors and new contributors.

Reply:
Not auto-generated.  If the numbers are not discussed and reanalyzed in the code review process they will become inaccurate.  This is the advantage of adding the docstrings instead of just `# noqa: C901, PLR0912, PLR0915`.  Experience tells me than any developer that lowers these numbers will proudly update the docstring.  If developers and code reviewers allow the numbers to rise then they will not modify the numbers but they will still be a strong signal to their more diligent successors.

These numbers can be easily recalculate by explicitly overriding the config file. `ruff check --select=C901,PLR091`

Reply:
OK, since there are only a few of these I'm ok adding them.

Reply:
Should we had to this "Run `ruff` with the above `noqa` comment removed to update these numbers".  I fear that without such a hit these comments will never get updated.